Explorer/Assets/DCL/Profiles/RealmProfileRepository.cs

```diff
@@ -0,0 +1,203 @@
+using CommunicationData.URLHelpers;
+using Cysharp.Threading.Tasks;
+using DCL.WebRequests;
+using ECS;
+using Ipfs;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace DCL.Profiles
+{
+    public class RealmProfileRepository : IProfileRepository
+    {
+        private readonly IWebRequestController webRequestController;
+        private readonly IRealmData realm;
+        private readonly IProfileCache profileCache;
+        private readonly URLBuilder urlBuilder = new ();
+
+        public RealmProfileRepository(IWebRequestController webRequestController,
+            IRealmData realm,
+            IProfileCache profileCache)
+        {
+            this.webRequestController = webRequestController;
+            this.realm = realm;
+            this.profileCache = profileCache;
+        }
+
+        public async UniTask<Profile?> GetAsync(string id, int version, CancellationToken ct)
+        {
+            if (string.IsNullOrEmpty(id)) return null;
+
+            IIpfsRealm ipfs = realm.Ipfs;
+
+            urlBuilder.Clear();
+
+            urlBuilder.AppendDomain(ipfs.LambdasBaseUrl)
+                      .AppendPath(URLPath.FromString($"profiles/{id}"))
+                      .AppendParameter(new URLParameter("version", version.ToString()));
+
+            URLAddress url = urlBuilder.Build();
+
+            try
+            {
+                GenericGetRequest response = await webRequestController.GetAsync(new CommonArguments(url), ct);
+
+                using GetProfileJsonRootDto root = await response.CreateFromNewtonsoftJsonAsync<GenericGetRequest, GetProfileJsonRootDto>(
+                    createCustomExceptionOnFailure: (exception, text) => new ProfileParseException(id, version, text, exception),
+                    converters: new ProfileJsonRootDtoConverter());
+
+                if (root.avatars == null) return null;
+                if (root.avatars.Count == 0) return null;
+
+                // TODO: probable responsibility issues thus we might not want to affect the cache
+                // but avoids extra allocations in case the profile already exists
+                Profile profile = profileCache.Get(id) ?? new Profile();
```

Comment:
Creating an empty `Profile` provokes all fields to be nullable. Despite they are set in the process, many warnings appears in compilation time. Perhaps we can find a workaround.